yeonjae/programmers-2/할인행사.js

```diff
@@ -0,0 +1,32 @@
+function solution(want, number, discount) {
+  let result = 0;
+
+  discount.forEach((v, i) => {
+    let temp = [...discount].slice(i, i + 10);
+    let cnt = 0;
+
+    if (temp.length < 10) {
+      return 0;
+    }
+
+    for (let j = 0; j < want.length; j++) {
+      if (temp.filter((el) => el == want[j]).length == number[j]) {
+        cnt++;
+      }
+    }
+
+    if (cnt == want.length) {
+      result++;
+    }
+  });
+
+  return result;
+}
```

Comment:
```suggestion
function solution(want, number, discount) {
  let result = 0;

  for (let i = 0; i < discount.length; i++) {
    const slicedDiscount = discount.slice(i, i + 10);
    let cnt = 0;

    if (slicedDiscount.length < 10) break;

    for (let j = 0; j < want.length; j++) {
      const isValid =
        slicedDiscount.filter((el) => el === want[j]).length === number[j];

      if (isValid) cnt++;
    }
    if (cnt === want.length) {
      result++;
    }
  }
  return result;
}
```
- 풀이잘봤습니다 고생하셨습니다. 조금 아쉬운 부분이 많은데요.. 🥲
- forEach문에서 `return`문이 조금 어색하네요. return으로 `0`을 반환하는 것도 의도가 어떤건지 모르겠습니다 🥲 아마 일반적인 for문에서 `break`문을 의도하신 거겠죠?? 🤔 목적에 맞게 사용하는게 좋을 것 같습니다~~
- `==`연산자를 사용하셨습니다 `===` 연산자로 사용해주세요~ (이유는 당연히 아실거라 생각이 드는데 검토가 안되신건지 ㅠ)
- slice는 애초에 새로운 배열을 반환하기 때문에 굳이 전개 연산자로 복사해서 사용하실 이유가 없습니다~ 🙇‍♂️
- if문의 복잡한 조건은 isValid(임시 네이밍)으로 분리시켜줬습니다 (이것도 이유는 아실거라 생각합니다!!)

<br />

- 검토 좀 만 더 신경써주세요 ㅠㅠ